### PR TITLE
feat(core): resolve webcrypto from node:crypto for Node18

### PR DESCRIPTION
**Pull request by HuiSF**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — DISMISSED** [review]

File: .github/actions/load-verdaccio-with-amplify-js/action.yml [hunk 1]
```diff
@@ -18,25 +19,32 @@ runs:
...
+        -H "Accept: application/json" \
+        -H "Content-Type:application/json" \
+        -X PUT --data "{\"name\": \"$NPM_USER\", \"password\": \"$NPM_PASS\", \"email\": \"$NPM_EMAIL\"}" \
+        $NPM_REGISTRY/-/user/org.couchdb.user:$NPM_USER 2>&1 | jq -r '.token')
+
+        # Set the Verdaccio registry and set the token for logging in
+        yarn config set registry $NPM_REGISTRY
+        npm set registry $NPM_REGISTRY
+        npm set //"$NPM_REGISTRY_HOST"/:_authToken $TOKEN
     - name: Configure registry and git
```
Comment (jimblanc) [inline .github/actions/load-verdaccio-with-amplify-js/action.yml:39]: Nit: Could update description and remove unneeded env params

File: packages/aws-amplify/package.json [hunk 2]
```diff
@@ -288,6 +288,9 @@
 	"devDependencies": {
 		"typescript": "5.0.2"
 	},
+	"engines": {
```
Comment (jimblanc) [inline packages/aws-amplify/package.json:291]: Sanity check: Any risk of breakage here? What happens in NPM & yarn if someone is using an older version?

**Review by jimblanc — COMMENTED** [review]
Looks good to me, few comments

File: packages/aws-amplify/package.json (hunk 2 above)
Comment (HuiSF) [reply packages/aws-amplify/package.json]: As adding the node version restriction is considered as a breaking change, removing it from this PR. When the library is used in a Node.js environment with version < 16, an error will be thrown as unable to resolve the Web crypto API.

**Review by jimblanc — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]

**Comment by haverchuck** [comment]
@HuiSF Does this impact people who are using SSR? Does crypto change get used during token refresh?

**Comment by HuiSF** [comment]
> @HuiSF Does this impact people who are using SSR? Does crypto change get used during token refresh?

Hey @haverchuck ,

The Webcrypto APIs are not used within token refresh flows on the server side.

On the server side of a SSR framework, Webcrypto API will be resolved from Node.js (it's already happening before this change with Node.js v20+). After this change, the Webcrypto API is resolvable with a Node.js version that provides Webcrypto. Currently, there is no Amplify server-side APIs consuming Webcrypto APIs.

**Pull request merged**